src/Switch/index.tsx

```diff
@@ -0,0 +1,192 @@
+    size === "normal" ? 2 : size === "large" ? 3 : assertUnreachable(size);
+
+  return (
+    <ClassNames>
+      {({ css }) => (
+        <Label className={className} style={style}>
+          <VisuallyHidden>
+            <input {...inputProps} {...focusProps} ref={ref} />
+          </VisuallyHidden>
+
+          <div
+            className={css({
+              flex: 1,
+              marginRight: showTextualState
+                ? 12
+                : size === "large"
+                ? 30
+                : size === "normal"
+                ? undefined
+                : assertUnreachable(size),
+            })}
+          >
+            {props.children}
+          </div>
+
+          {showTextualState && (
+            <div
+              aria-hidden
+              className={css({
+                color:
+                  theme === "light"
+                    ? props.isDisabled
+                      ? undefined
+                      : state.isSelected
+                      ? undefined
+                      : colors.grey.base
+                    : theme === "dark"
+                    ? props.isDisabled
+                      ? undefined
+                      : state.isSelected
+                      ? colors.white
+                      : colors.midnight.lighter
+                    : assertUnreachable(theme),
+                fontWeight: state.isSelected ? 600 : 400,
+                marginRight: 8,
+              })}
+            >
+              {state.isSelected ? "ON" : "OFF"}
+            </div>
+          )}
+
+          <div
+            aria-hidden
+            key={props.isDisabled ? "disabled" : "enabled"}
+            className={css({
+              backgroundColor: state.isSelected
+                ? getOffsetInPalette(props.isDisabled ? 2 : 0, "lighter", color)
+                : props.isDisabled
+                ? colors.silver.dark
+                : colors.grey.light,
+              borderRadius: dotSize / 2 + borderSize,
+              boxShadow: [
+                isFocusVisible && `0 0 0 2px ${colors.blue.lighter}`,
+                !props.isDisabled && "inset 0 0 1px 0 rgba(18, 21, 26, 0.4)",
+              ]
+                // The generic attached to the filter return type will indicate
+                // to TypeScript that we're stripping out all non-strings,
+                // meaning the booleans.
+                .filter((value): value is string => !!value)
+                .join(", "),
+              cursor: !props.isDisabled ? "pointer" : undefined,
+              height: dotSize + borderSize * 2,
+              padding: borderSize,
+              position: "relative",
+              width: (8 / 3) * dotSize,
+            })}
+          >
+            <motion.div
+              animate={{
+                x: state.isSelected ? dotSize + borderSize * 2 : 0,
+              }}
+              initial={false}
+              transition={{ duration: 0.5, type: "tween", ease: "easeOut" }}
```

Comment:
Can we speed this up a little, say 150ms?
```suggestion
              transition={{ duration: 0.15, type: "tween", ease: "easeOut" }}
```